theano/tensor/nnet/abstract_conv.py

```diff
@@ -301,6 +301,12 @@ def grad(self, inp, grads):
                                                self.filter_flip)(
 
             bottom, top, weights.shape[-2:])
+
+        # Make sure that the broadcastable pattern of the inputs is used
+        # for the gradients, even if the grad opts are not able to infer
+        # that the dimensions are broadcastable.
+        d_bottom = patternbroadcast(d_bottom, bottom.broadcastable)
```

Comment:
The legacy convolution on the CPU do not add extra node in the graph. You can do this change directly in the make_node of AbstractConv2d_gradWeights.

bottom is the first input of the make_node, so the information is there for d_bottom. So remove it.

for d_weights, you need this extra node in the graph with this method or pass an extra parameter in the make_node. I prefer to add a parameter to make_node, but if you prefer the extra node, I won't oppose it.

Idem elsewhere.